### PR TITLE
feat: flair encoder

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
https://github.com/jina-ai/internal-tasks/issues/92

- [x] improve the readme
- [x] add more tests

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,14 @@
-FROM jinaai/jina:master as base
+FROM jinaai/jina:2.0.0rc6
```
Comment (florian-hoenicke) [inline Dockerfile:1]: We can use a cuda base image to let it run on gpu

File: flair_text.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+from typing import Union, Tuple, List, Any, Iterable
+
+import numpy as np
+import torch
+from jina import Executor, requests, DocumentArray
+
+
+def _batch_generator(data: List[Any], batch_size: int):
+    for i in range(0, len(data), batch_size):
+        yield data[i:min(i + batch_size, len(data))]
```
Comment (davidbp) [inline flair_text.py]: No need to write `min`. you can set this as `data[i: i + batch_size]` the `min` is done at `C` level faster and much more efficient.

File: flair_text.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            document_batches_generator = self._get_input_data(docs, parameters)
+            self._create_embeddings(document_batches_generator)
+
+    def _create_embeddings(self, document_batches_generator: Iterable):
+        for document_batch in document_batches_generator:
+            text_batch = [d.text for d in document_batch]
+            text_batch = np.array(text_batch)
+
+            from flair.data import Sentence
+            c_batch = [Sentence(row) for row in text_batch]
```
Comment (davidbp) [inline flair_text.py]: Ins't this code
```
 text_batch = [d.text for d in document_batch]
text_batch = np.array(text_batch)
from flair.data import Sentence
c_batch = [Sentence(row) for row in text_batch]
```
Equivalent to this ?
```
from flair.data import Sentence
c_batch = [Sentence(d.text) for d in document_batch]
```
You would avoid creating a list and a numpy array this way

File: flair_text.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            self.model.embed(c_batch)
+            embedding_batch = [self.tensor2array(c_text.embedding) for c_text in c_batch]
+            for document, embedding in zip(document_batch, embedding_batch):
+                document.embedding = embedding
```
Comment (davidbp) [inline flair_text.py]: Here also embedding_batch does not seem very useful:
```
embedding_batch = [self.tensor2array(c_text.embedding) for c_text in c_batch]
for document, embedding in zip(document_batch, embedding_batch):
      document.embedding = embedding
```
Could be done as
```
 for document, c_text in zip(document_batch, c_batch):
         document.embedding = self.tensor2array(self.tensor2array(c_text.embedding))
```

File: flair_text.py (hunk 3 above)
Comment (Yongxuanzhang) [reply flair_text.py]: yes

File: flair_text.py (hunk 4 above)
Comment (Yongxuanzhang) [reply flair_text.py]: thanks a lot!

**Review by CatStark — APPROVED** [review]
LGTM

File: manifest.yml [hunk 5]
```diff
@@ -1,12 +1,10 @@
...
+name: FlairTextEncoder
+kind: encoder
+description: ImagePaddlehubEncoder encodes text data.
 author: Jina AI Dev-Team (dev-team@jina.ai)
 url: https://jina.ai
 vendor: Jina AI Limited
-documentation: https://github.com/jina-ai/jina-hub
 version: 0.0.1
 license: apache-2.0
-keywords: [image, normalization, reader, resize]
```
Comment (Kelton8Z) [inline manifest.yml:12]: I changed the keywords of nlp to text, and torch to PyTorch to conform with our 2.0 executor guideline

Comment (Yongxuanzhang) [reply manifest.yml:12]: thanks a lot!

**Pull request merged**